src/pydocstyle/config.py

```diff
@@ -208,7 +208,10 @@ def _get_config(self, node):
         if path in self._cache:
             return self._cache[path]
 
-        config_file = self._get_config_file_in_folder(path)
+        if self._options.config_file:
```

Comment:
IMO, specifying a custom directory path (rather than a specific config file) should be a different command-line option (perhaps `--config-dir`?) and is outside the scope of this change.